pkg/virt-handler/vm.go

```diff
@@ -680,6 +684,11 @@ func (d *VirtualMachineController) removeCondition(vm *v1.VirtualMachine, cond v
 
 func (d *VirtualMachineController) setVmPhaseForStatusReason(domain *api.Domain, vm *v1.VirtualMachine) {
 	vm.Status.Phase = d.calculateVmPhaseForStatusReason(domain, vm)
+	// If the VM is finally down and we were dealing with  a graceful delete, remove the finalizer, to allow final deletion
+	if vm.IsFinal() && vm.ObjectMeta.DeletionTimestamp != nil && len(vm.ObjectMeta.Finalizers) > 0 && vm.ObjectMeta.Finalizers[0] == v1.FinalizerDeleteVirtualMachine {
```

Comment:
This is probably just temporary for WIP, but just in case it isn't I don't feel great about this location for this finalizer check. Also, we should write a function that checks/removes the finalizer from the list without assuming index 0.